Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2860 - Deploy Redirect to ephemeral environment

# Deploy Redirect to Ephemeral Environment

Resolves [EREGCSC-2860](https://jiraent.cms.gov/browse/EREGCSC-2860)

## Description
This pull request changes:
- Implements automated deployment of Redirect API to ephemeral environments for PR testing
- Adds GitHub Actions workflow for PR-specific stack deployment
- Includes automatic cleanup when PR is closed

## Steps to manually verify this change:
1. Open a new PR
   - Verify GitHub Action triggers automatically
   - Check CDK deployment logs for stack creation

2. Test deployed endpoint
   - Find API endpoint URL in aws console 
   - Verify API responds as expected
   - Check AWS Console for PR-specific resources (format: cms-eregs-eph-{pr-number}-redirect-api)

3. Close PR to verify cleanup
   - Observe cleanup workflow trigger
   - Verify stack is destroyed
   - Check AWS Console to confirm resource removal

## Comments
1. Stack pattern: cms-eregs-eph-{pr-number}-redirect-api
2. Ephemeral resources auto-destroy on PR close
3. Debug logs available with CDK_DEBUG=true
